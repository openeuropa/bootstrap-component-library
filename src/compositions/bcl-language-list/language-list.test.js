import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-language-list/data/data--neutral.js";
import demoDataEU from "@openeuropa/bcl-language-list/data/data--eu.js";
import demoDataEC from "@openeuropa/bcl-language-list/data/data--ec.js";

const template = "@oe-bcl/bcl-language-list/language-list.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Language", () => {
  test("Neutral renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test("EU renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoDataEU)).resolves.toMatchSnapshot();
  });

  test("EC renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoDataEC)).resolves.toMatchSnapshot();
  });
});
