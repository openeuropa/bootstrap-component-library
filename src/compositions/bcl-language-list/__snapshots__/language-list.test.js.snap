// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Language EC renders correctly 1`] = `
<jest>
  <div
    class="bcl-language-list bcl-language-list--eu pt-5"
  >
    <div
      class="container"
    >
      <div
        class="text-center"
      >
        <div
          class="d-inline-flex align-items-center"
        >
          <img
            alt="logo"
            class="w-100 mb-3"
            src="https://cdn1.fpfis.tech.ec.europa.eu/ecl/v3.0.2/ec/images/logo/logo-ec--en.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="main pt-4"
    >
      <div
        class="container"
      >
        <div
          class="pb-2"
        >
          <h5
            class="mb-4"
          >
            EU official languages
          </h5>
          <div
            class="row"
          >
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                български
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                español
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                čeština
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                dansk
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Deutsch
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                eesti
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                ελληνικά
              </button>
              <button
                class="selected justify-content-between w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                English
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#check-lg"
                  />
                </svg>
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                français
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Gaeilge
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                hrvatski
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                italiano
              </button>
            </div>
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                latviešu
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                lietuvių
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                magyar
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Malti
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Nederlands
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                polski
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                português
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                română
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                slovenčina
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                slovenščina
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                suomi
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                svenska
              </button>
            </div>
          </div>
        </div>
        <div
          class="pb-2"
        >
          <h5
            class="mb-4"
          >
            Other languages
          </h5>
          <div
            class="row"
          >
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                中文
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Türk
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                pусский
              </button>
            </div>
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Català
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                عَرَبِيّ
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Language EU renders correctly 1`] = `
<jest>
  <div
    class="bcl-language-list bcl-language-list--eu pt-5"
  >
    <div
      class="container"
    >
      <div
        class="text-center"
      >
        <div
          class="d-inline-flex align-items-center"
        >
          <img
            alt="logo"
            class="w-100"
            src="https://cdn1.fpfis.tech.ec.europa.eu/ecl/v3.0.2/eu/images/logo/logo-eu--mute.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="main pt-4"
    >
      <div
        class="container"
      >
        <div
          class="pb-2"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                български
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                español
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                čeština
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                dansk
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Deutsch
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                eesti
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                ελληνικά
              </button>
              <button
                class="selected justify-content-between w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                English
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#check-lg"
                  />
                </svg>
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                français
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Gaeilge
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                hrvatski
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                italiano
              </button>
            </div>
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                latviešu
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                lietuvių
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                magyar
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Malti
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Nederlands
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                polski
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                português
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                română
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                slovenčina
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                slovenščina
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                suomi
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                svenska
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Language Neutral renders correctly 1`] = `
<jest>
  <div
    class="bcl-language-list bcl-language-list--neutral pt-5"
  >
    <div
      class="container"
    >
      <div
        class="d-inline-flex align-items-center"
      >
        <img
          alt="logo"
          src="https://inno-ecl.s3.amazonaws.com/media/examples/placeholder.svg"
        />
        <h5
          class="h5 ms-4"
        >
          Project name
        </h5>
      </div>
    </div>
    <div
      class="main pt-4"
    >
      <div
        class="container"
      >
        <div
          class="pb-2"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                български
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                español
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                čeština
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                dansk
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Deutsch
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                eesti
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                ελληνικά
              </button>
              <button
                class="selected justify-content-between w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                English
                <svg
                  class="ms-2-5 bi icon--fluid"
                >
                  <use
                    xlink:href="/icons.svg#check-lg"
                  />
                </svg>
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                français
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Gaeilge
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                hrvatski
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                italiano
              </button>
            </div>
            <div
              class="col-12 col-md"
            >
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                latviešu
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                lietuvių
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                magyar
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Malti
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                Nederlands
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                polski
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                português
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                română
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                slovenčina
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                slovenščina
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                suomi
              </button>
              <button
                class="w-100 mb-2-5 justify-content-start btn btn-primary btn-md"
                type="button"
              >
                svenska
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
